src/connectivity.js

```diff
@@ -0,0 +1,185 @@
+      responseData.result = "Passed";
+    }
+
+    response.on('data', function (chunk) {
+      responseData.data.push(chunk);
+    });
+
+    response.on('end', function () {
+      responseData.data = Buffer.concat(responseData.data).toString();
+      callback(responseData);
+    });
+
+  });
+
+  request.on('error', function (err) {
+    responseData.errorMessage = err.toString();
+    callback(responseData);
+  });
+
+  request.setTimeout(constants.CONNECTIVITY_REQ_TIMEOUT, function () {
+    request.destroy("Request Timed Out");
+  });
+
+  request.end();
+};
+
+
+var ConnectivityChecker = {
+
+  connectionChecks: [],
+
+  reqOpsWithoutProxy: function () {},
+  reqOpsWithProxy: function () {},
+
+  httpToHubWithoutProxy: function (callback) {
+    var requestUrl = constants.HUB_STATUS_URL;
+    var requestOptions = ConnectivityChecker.reqOpsWithoutProxy(requestUrl, 'http');
+    fireRequest(requestOptions, 'http', 'HTTP Request To Hub Without Proxy', [200], function (response) {
+      callback(response);
+    });
+  },
+
+  httpToRailsWithoutProxy: function (callback) {
+    var requestUrl = constants.RAILS_AUTOMATE;
+    var requestOptions = ConnectivityChecker.reqOpsWithoutProxy(requestUrl, 'http');
+    fireRequest(requestOptions, 'http', 'HTTP Request To Rails Without Proxy', [200, 301], function (response) {
+      callback(response);
+    });
+  },
+
+  httpsToHubWithoutProxy: function (callback) {
+    var requestUrl = constants.HUB_STATUS_URL;
+    var requestOptions = ConnectivityChecker.reqOpsWithoutProxy(requestUrl, 'https');
+    fireRequest(requestOptions, 'https', 'HTTPS Request To Hub Without Proxy', [200], function (response) {
+      callback(response);
+    });
+  },
+
+  httpsToRailsWithoutProxy: function (callback) {
+    var requestUrl = constants.RAILS_AUTOMATE;
+    var requestOptions = ConnectivityChecker.reqOpsWithoutProxy(requestUrl, 'https');
+    fireRequest(requestOptions, 'https', 'HTTPS Request to Rails Without Proxy', [301, 302], function (response) {
+      callback(response);
+    });
+  },
+
+  httpToHubWithProxy: function (callback) {
+    var requestUrl = constants.HUB_STATUS_URL;
+    var requestOptions = ConnectivityChecker.reqOpsWithProxy(requestUrl, 'http');
+    fireRequest(requestOptions, 'http', 'HTTP Request To Hub With Proxy', [200], function (response) {
+      callback(response);
+    });
+  },
+
+  httpToRailsWithProxy: function (callback) {
+    var requestUrl = constants.RAILS_AUTOMATE;
+    var requestOptions = ConnectivityChecker.reqOpsWithProxy(requestUrl, 'http');
+    fireRequest(requestOptions, 'http', 'HTTP Request To Rails With Proxy', [301], function (response) {
+      callback(response);
+    });
+  },
+
+
+  /**
+   * Decides the checks to perform based on whether any proxy is provided by the
+   * user or not.Along with the checks, it builds the request options template
+   * required to fire the connectivity check requests.
+   */
+  decideConnectionChecks: function () {
```

Comment:
Inclusion of basicAuth checks?

Reply:
These checks are just to see if the user's machine is able to reach BrowserStack Hub & Rails or not.
Where would the basic auth come into picture?

Reply:
`api.browserstack.com`?

Reply:
The current rails URL is enough to check if the user's machine can reach BrowserStack or not. 
It was a conscious call to not depend on the creds for checking connectivity with Rails, thus, not going with `api.bro...`

Reply:
If the user is running JSTesting and using some framework like karma or TestCafe and connectivity for the `api.browserstack.com` is not working from the proxy, then how are we fetching that data? Should we warn the user before running the session that it is not accessible from their environment or what?

Reply:
Resolved F2F